File: AdministracionDeTienda/src/gestorAplicacion/clasesBase/Persona.java
package gestorAplicacion.clasesBase;
import java.util.ArrayList;
import java.io.Serializable;
import java.util.HashSet;
import java.util.Random;
import java.util.Set;

public abstract class Persona implements Comparable<Persona>,Serializable{
    private int calificacion;
    private String nombre;
    private CuentaBancaria cuenta;
    
    public Persona() {
    	
    }

    public Persona(String nombre, int calificacion) {
		this.calificacion=calificacion;
		this.nombre=nombre;
	}
    public Persona(String nombre, int calificacion, CuentaBancaria cuenta){
      this(nombre,calificacion);
      this.cuenta=cuenta;
      cuenta.getEntidad().getHistorialesCrediticios().put(cuenta,new ArrayList<Credito>());
      cuenta.setPropietario(this);
    }

   

	public abstract int calificar();

    public static ArrayList<Producto> generarProductos(ArrayList<Producto> listaProductos) {
        ArrayList<Producto> productosSeleccionados = new ArrayList<>();
        Random random = new Random();
        Set<Integer> indicesSeleccionados = new HashSet<>();

        int cantidadProductos = listaProductos.size();
        int cantidadSeleccionada = random.nextInt(cantidadProductos + 1);

        while (cantidadSeleccionada > 0) {
            int indiceAleatorio = random.nextInt(cantidadProductos);
            if (!indicesSeleccionados.contains(indiceAleatorio)) {
                Producto productoSeleccionado = listaProductos.get(indiceAleatorio);
                productosSeleccionados.add(productoSeleccionado);
                indicesSeleccionados.add(indiceAleatorio);
                cantidadSeleccionada--;
            }
        }
        return productosSeleccionados;
    }

        public CuentaBancaria getCuenta(){
            return cuenta;
        }

        public void setCuenta(CuentaBancaria cuenta){
            this.cuenta = cuenta;
        }

        public String  getNombre(){
            return nombre;
        }

        public void setNombre(String nombre){
            this.nombre = nombre;
        }

        public int getCalificacion(){
            return calificacion;
        }

        public void setCalificacion(int calificacion){
            this.calificacion = calificacion;
        }

        public String toString (){
          return nombre;
        }


        public String demandar() {
      		return toString()+" ha decidido no demandarlo.\n";
      	}

      




      	@Override
      	public int compareTo(Persona otra) {
      	//La mejor oferta es aquella con menor total
      	//vengo antes que la otra si es NEGATIVO---ORDENA DE MAYOR A MENOR
      		return (int)(otra.getCalificacion()-this.calificacion);
      	}



      
}

File: Python/src/gestorAplicacion/clasesBase/Persona.py
from abc import ABC, abstractmethod
import random
import pickle


class Persona(ABC):
    def __init__(self, nombre, calificacion, cuenta):
        self.calificacion = calificacion
        self.nombre = nombre
        self.cuenta = cuenta
    
    @abstractmethod
    def valorCalificacion(self, *args):
        pass

    @staticmethod
    def generarProductos(lista_productos):
        productosSeleccionados = []
        indices_seleccionados = set()
        cantidad_productos = len(lista_productos)
        cantidad_seleccionada = random.randint(0, cantidad_productos)

        while cantidad_seleccionada > 0:
            indice_aleatorio = random.randint(0, cantidad_productos - 1)
            if indice_aleatorio not in indices_seleccionados:
                producto_seleccionado = lista_productos[indice_aleatorio]
                productosSeleccionados.append(producto_seleccionado)
                indices_seleccionados.add(indice_aleatorio)
                cantidad_seleccionada -= 1

        return productosSeleccionados

    def getCuenta(self):
        return self.cuenta

    def setCuenta(self, cuenta):
        self.cuenta = cuenta

    def getNombre(self):
        return self.nombre

    def setNombre(self, nombre):
        self.nombre = nombre

    def getCalificacion(self):
        return self.calificacion

    def setCalificacion(self, calificacion):
        self.calificacion = calificacion

    def __str__(self):
        return self.nombre

    def demandar(self):
        return f"{str(self)} ha decidido no demandarlo.\n"

    def __lt__(self, other):
        return self.calificacion > other.calificacion